lib/ansible/plugins/action/net_template.py

```diff
@@ -93,6 +93,16 @@ def _handle_template(self):
         except IOError:
             return dict(failed=True, msg='unable to load src file')
 
+        # Create a template search path in the following order:
+        # [working_path, _role_path, *_block._role_paths, dirname(source)]
+        searchpath = [working_path]
+        if self._task._role is not None:
+            searchpath.append(self._task._role._role_path)
+            if self._task._block._dep_chain:
+                for role in sorted(self._task._block.get_dep_chain()):
```

Comment:
Thank you @Qalthos!  This is suggested by @bcoca while back and I just followed his way. :)

In any case, I'll quickly check the code and get back to you, either removing the sorted() or comment here.

Cheers,